pkg/tests/api/alerting/api_alertmanager_silence_test.go

```diff
@@ -0,0 +1,414 @@
+		DisableLegacyAlerting: true,
+		EnableUnifiedAlerting: true,
+		DisableAnonymous:      true,
+		AppModeProduction:     true,
+	})
+
+	grafanaListedAddr, env := testinfra.StartGrafanaEnv(t, dir, path)
+
+	createUser(t, env.SQLStore, env.Cfg, user.CreateUserCommand{
+		DefaultOrgRole: string(org.RoleAdmin),
+		Password:       "admin",
+		Login:          "admin",
+	})
+
+	adminApiClient := newAlertingApiClient(grafanaListedAddr, "admin", "admin")
+
+	// Create the namespace we'll save our alerts to.
+	f1 := folder.Folder{
+		UID:   util.GenerateShortUID(),
+		Title: "Folder 1",
+	}
+	adminApiClient.CreateFolder(t, f1.UID, f1.Title)
+	f2 := folder.Folder{
+		UID:   util.GenerateShortUID(),
+		Title: "Folder 2",
+	}
+	adminApiClient.CreateFolder(t, f2.UID, f2.Title)
+
+	group1 := generateAlertRuleGroup(1, alertRuleGen())
+	group2 := generateAlertRuleGroup(1, alertRuleGen())
+
+	respModel, status, _ := adminApiClient.PostRulesGroupWithStatus(t, f1.UID, &group1)
+	require.Equal(t, http.StatusAccepted, status)
+	ruleInFolder1UID := respModel.Created[0]
+	respModel, status, _ = adminApiClient.PostRulesGroupWithStatus(t, f2.UID, &group2)
+	require.Equal(t, http.StatusAccepted, status)
+	ruleInFolder2UID := respModel.Created[0]
+
+	type silenceAction string
+	const (
+		readSilence   silenceAction = "read"
+		createSilence silenceAction = "create"
+		updateSilence silenceAction = "update"
+		deleteSilence silenceAction = "delete"
+	)
+
+	type silenceType string
+	const (
+		generalSilence       silenceType = "generalSilence"
+		ruleSilenceInFolder1 silenceType = "ruleSilenceInFolder1"
+		ruleSilenceInFolder2 silenceType = "ruleSilenceInFolder2"
+	)
+
+	silenceGens := map[silenceType]func() ngmodels.Silence{
+		generalSilence:       ngmodels.SilenceGen(),
+		ruleSilenceInFolder1: ngmodels.SilenceGen(ngmodels.SilenceMuts.WithMatcher(models.RuleUIDLabel, ruleInFolder1UID, labels.MatchEqual)),
+		ruleSilenceInFolder2: ngmodels.SilenceGen(ngmodels.SilenceMuts.WithMatcher(models.RuleUIDLabel, ruleInFolder2UID, labels.MatchEqual)),
+	}
+
+	defaultStatus := map[silenceAction]map[bool]int{
+		updateSilence: {true: http.StatusAccepted, false: http.StatusForbidden},
+		deleteSilence: {true: http.StatusOK, false: http.StatusForbidden},
+		createSilence: {true: http.StatusAccepted, false: http.StatusForbidden},
+		readSilence:   {true: http.StatusOK, false: http.StatusForbidden},
+	}
+
+	testCases := []struct {
+		name             string
+		orgRole          org.RoleType // default RoleNone
+		permissions      []accesscontrol.Permission
+		defaultAllowed   bool                                  // Default allowed/forbidden for actions not in statusExceptions.
+		statusExceptions map[silenceType]map[silenceAction]int // Exceptions to defaultAllowed.
+		listContents     []silenceType                         // nil = forbidden.
+	}{
+		// OSS Builtins
+		{
+			name:    "No permissions",
+			orgRole: org.RoleNone,
```

Comment:
This role is not permitted in OSS

Reply:
Ah, good catch. Forgot that detail, will update

Reply:
b4131d342959fee41ed7e181d2e2314533b9f9a5